### DISCUSSION TITLE
9-location Universal Core design- couldn't get Tab work

### DISCUSSION THREAD

**Discussion by su-aac** [UNC-Project-Open-AAC/OS-DPI#229] [Help]
Hello,
I am replicating the 9-location Universal Core design on OSDPI and need help to figure out how to make all symbols shown on each page.
The 9-location Universal Core design look like this:
[9_Location_Book_091216.pdf](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/13300674/9_Location_Book_091216.pdf)

The symbols only show on page 3 but not on other pages. I also want to delete the message bar on top.
<img width="956" alt="image" src="https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123387291/4acde946-c289-480a-b42d-3e3c12ea25e4">
<img width="948" alt="image" src="https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123387291/f50fa04d-b2e9-42c5-becb-8718aef8f4d5">

I attached my design here. Thank you so much!!!
[Project_Core_9x4.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/13300652/Project_Core_9x4.zip)

**Comment by gbishop** [comment]
You have each grid filtering on sheetName and role but you never update role in your actions. As written, after fixing your content so the first column on sheet 1 is named role, the first grid works. My guess is the others do also, if you update the $role variable appropriately.

**Comment by sofia-bg** [comment]
@hsiaotingsu since the purpose is just to print i got rid of #role entirely and it works. I also got rid of the speech display bar and "hid" the tabs. 

[Project_Core_9x4_SBG.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/13359218/Project_Core_9x4_SBG.zip)

To make it look more like the original I added the following to a customize tab: 

    button[#label] { font-size: 50px; margin-left: 5px; margin-bottom: 5px; margin-right: 5px; margin-top: 5px; border-radius: 0%;}

This makes the font size bigger, the margin information is what spaces the buttons from each other and the border radius changes the corners from being rounded to being right angles. 

@gbishop what I cannot figure out how to do is change so that the font is above the image. Using the various padding codes just changes the relative size of the symbols, and  text-align doesn't seem to change the relative placement of the text either... 

<img width="300" alt="Screenshot 2023-11-14 at 8 47 14 PM" src="https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/92817124/b3a565b7-5996-4149-99ba-b49bca798012">

**Reply by sofia-bg** [reply]
@lageist if you know the font that you all used we can use customize to change the font to match as well

**Reply by gbishop** [reply]
@sofia-bg Move the captions to the top with:

button figure { flex-flow: column-reverse; }

**Comment by gbishop** [comment]
I don't understand this design. How do you change tabs? Manually?

And why are you using a tab control. A single Grid could do the same thing. Just filter on a variable that is the "page" you want.

**Reply by sofia-bg** [reply]
yes this isn't intended to be a high tech design but a way to quickly change out symbols for printed versions as we try to make available versions with a variety of skin tones represented. So can just manually make the tabs viewable, change tab, and then print.

**Comment by su-aac** [comment]
Thank you both, @gbishop and @sofia-bg so much!!
<img width="475" alt="image" src="https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123387291/4ecbbfb2-0a75-4cc2-8a67-ed78ec50559f">
Now the layout looks pretty similar to the picture Sofia attached, and I can use the "Activate" button under each Tab to navigate through pages. This template will support me in creating other Universal Core communication systems formats. I really appreciate your help!
[Project_Core_9x4_SBG_HS.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/13428289/Project_Core_9x4_SBG_HS.zip)

**Comment by gbishop** [comment]
Glad to help!
